fe/fe-core/src/main/java/com/starrocks/connector/delta/CachingDeltaLakeMetadata.java

```diff
@@ -0,0 +1,507 @@
+                            .createDeltaCacheBdbDao(tbl, METADATA_CACHE_DISK_ROOT_PATH);
+                    cacheDeltaLocalCacheDao.put(tbl, deltaCacheBdbDao);
+                }
+            }
+        }
+        return deltaCacheBdbDao;
+    }
+
+    public DeltaTableMetadata loaderCacheTable(DeltaLakeTableName tbl) {
+
+        return getMetadataDiskCacheDao(tbl).getMetadata();
+    }
+
+    private static CacheBuilder<Object, Object> newCacheBuilder(long expiresAfterWriteSec,
+                                                                long maximumSize) {
+        CacheBuilder<Object, Object> cacheBuilder = CacheBuilder.newBuilder();
+
+        if (expiresAfterWriteSec >= 0) {
+            cacheBuilder.expireAfterWrite(expiresAfterWriteSec, TimeUnit.SECONDS);
+        }
+        cacheBuilder.maximumSize(maximumSize);
+        return cacheBuilder;
+    }
+
+    /**
+     * Updating the local cache needs to be locked.
+     * @param snapshot snapshot
+     * @param tbl      tbl
+     */
+    public void refreshTable(Snapshot snapshot, DeltaLakeTableName tbl) throws Exception {
+        try {
+            long startTime = System.currentTimeMillis();
+            process(snapshot, tbl);
+            LOG.info("refresh table:{}'s metadata take total time:{} ms", tbl,
+                    (System.currentTimeMillis() - startTime));
+        } catch (Exception e) {
+            LOG.error("caching " + tbl.toString() + " error", e);
+            clearLocalCache(tbl);
+            throw new Exception("caching " + tbl + " error:" + e.getMessage());
+        }
+    }
+
+    private void clearLocalCache(DeltaLakeTableName tbl) {
+        cacheTableMetadata.invalidate(tbl);
+        cacheTablePartitions.invalidate(tbl);
+        cacheDeltaLocalCacheDao.invalidate(tbl);
+        snapshotVersionCheckTime.invalidate(tbl);
+    }
+
+    /**
+     * Persist first and then update the memory cache
+     */
+    private void process(Snapshot snapshot, DeltaLakeTableName tbl) throws RocksDBException {
+        Metadata metadata = snapshot.getMetadata();
+        long updateVersion = snapshot.getVersion();
+        List<AddFile> allFiles = snapshot.getAllFiles();
+        List<String> partitionColumns = metadata.getPartitionColumns();
+
+        String metadataJToJson = DeltaJsonUtil.convertMetadataJToJson(metadata);
+        DeltaTableMetadata tableMetadata = new DeltaTableMetadata(metadataJToJson, updateVersion);
+
+        //--- persistent partition group by addfile
+        Set<Map<String, String>> allPartionValues = Collections.newSetFromMap(new HashMap<>());
+        List<Pair<String, AddFile>> partitionedAddFiles = allFiles.stream().map(addFile -> {
+            // TODO SR currently does not support non-numeric cost calculations,
+            //  and the delta-connector does not implement push-down calculations based on statistics.
+            //  temporarily remove minValues first, maxValues, nullCount statistics, to reduce the local cache size
```

Comment:
Agreed. We have implemented statistics collection for table formats such as Hive and Iceberg. It's recommended to leverage SR's built-in statistic, so we don't need delta lake statistics. Furthermore, I think delta lake statistics lacks NDV which is very important for CBO.